### PR TITLE
fix: auth bypass

### PR DESCRIPTION
**Pull request by Sn0rt**
Compatible with original logic

### CODE REVIEW

File: lib/resty/jwt.lua [hunk 1]
```diff
@@ -549,7 +551,7 @@ function _M.sign(self, secret_key, jwt_obj)
     end
   end
 
-  if typ == str_const.JWE or jwt_obj.header.enc then
+  if jwt_obj.typ == str_const.JWE or (jwt_obj.typ == nil and (typ == str_const.JWE or jwt_obj.header.enc)) then
```
Comment (monkeyDluffy6017) [inline lib/resty/jwt.lua:554]: Does this need changed like below?

File: lib/resty/jwt.lua [hunk 2]
```diff
@@ -824,12 +826,17 @@ function _M.verify_jwt_obj(self, secret, jwt_obj, ...)
   end
 
   -- if jwe, invoked verify jwe
-  if jwt_obj[str_const.header][str_const.enc]  then
+  if jwt_obj.typ == str_const.JWE then
```
Comment (monkeyDluffy6017) [inline lib/resty/jwt.lua:829]: As you remove the check of `enc`, you should add the check in verify_jew_obj

File: lib/resty/jwt.lua (hunk 1 above)
Comment (monkeyDluffy6017) [inline lib/resty/jwt.lua:554]: Does the jwt_obj have `typ`?

Comment (Sn0rt) [reply lib/resty/jwt.lua:554]: image

Comment (monkeyDluffy6017) [reply lib/resty/jwt.lua:554]: The jwt_obj is not from the function `parse_jwt`

Comment (monkeyDluffy6017) [reply lib/resty/jwt.lua:554]: image
It's a jwt table we construct

Comment (Sn0rt) [reply lib/resty/jwt.lua:554]: image
So it is necessary to write this way to consider the use of compatible old APIs

Comment (Sn0rt) [reply lib/resty/jwt.lua:554]: image

File: lib/resty/jwt.lua (hunk 2 above)
Comment (Sn0rt) [reply lib/resty/jwt.lua:829]: I will fix this problem together in the next PR

**Review by monkeyDluffy6017 — APPROVED** [review]

**Pull request merged**